### PR TITLE
functionality to crop ICON grid and data 

### PR DESCRIPTION
**Pull request by cosunae**
Technical Description
-----------------------------

functionality to crop ICON grid and data. 
All full triangles (including its edges and vertices) whose cell center fall inside the lat/lon rectangle of the region of interest are included, meaning some edges / vertices with location outside of the region might be included

### CODE REVIEW

**Comment by cosunae** [comment]
launch jenkins

**Comment by cosunae** [comment]
launch jenkins

**Comment by victoria-cherkas** [comment]
Hi Carlos, apologies I have changed the flake8 config in the other documentation branch and would like this PR to conform to it: 

These two (check_loc and latlon_spec ) are perhaps not public so should be made private (if they are):

- iconarray/core/latlonhash.py:8:1: D103 Missing docstring in public function 
- iconarray/core/latlonhash.py:13:1: D101 Missing docstring in public class

Missing public module docstrings: 

- iconarray/core/crop.py:1:1: D100 Missing docstring in public module
- iconarray/core/latlonhash.py:1:1: D100 Missing docstring in public module 
- iconarray/tests/test_crop.py:1:1: D100 Missing docstring in public module

Can we add a brief desciption on more specifically what this test is testing?

- iconarray/tests/test_crop.py:6:1: D103 Missing docstring in public function

Let me know if I can help with these changes

File: iconarray/core/crop.py [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+        ...                   ...
+        inverse_flattening:   0.0
+        grid_level:           8
+        grid_root:            19
+        uuidOfParHGrid:       e6ddd597-9c90-27b1-fbac-c40d47f72ba0
+        uuidOfHGrid:          5a0a863d-2523-9515-7789-4930e3452bc0
+        global_grid:          0
+
+    open a dataset with data:
+    >>> ds_cell = xr.open_datasets("lfff00010000_cell.nc")
```
Comment (victoria-cherkas) [inline iconarray/core/crop.py:121]: xr.open_dataset without the s? 
Is it designed such that we should only pass data that only has one of the variable types - cell/edge/vertex to the Crop.__call__(ds) ?

**Comment by victoria-cherkas** [comment]
Hi Carlos I tried out the cropping but cant get it to work on my own data: /scratch/vcherkas/iconarray/scratch/testing_crop.ipynb

**Comment by cosunae** [comment]
launch jenkins

File: iconarray/core/crop.py (hunk 1 above)
Comment (cosunae) [reply iconarray/core/crop.py]: yes, true.

**Comment by cosunae** [comment]
launch jenkins

**Comment by cosunae** [comment]
launch jenkins

**Comment by cosunae** [comment]
I fixed the documentation issues you mentioned. However there are many more and I can not test/fix them since we have D1 ignored in the setup.cfg. 
I propose to activate D1 (or discuss the subset we want from D1) in a separate issue: 
https://github.com/C2SM/iconarray/issues/14

**Comment by cosunae** [comment]
launch jenkins

**Review by victoria-cherkas — APPROVED** [review]
Thanks for additional documentation!

**Pull request merged**